components/director/internal/domain/destination/service.go

```diff
@@ -374,6 +376,65 @@ func (s *Service) createOAuth2ClientCredentialsDestinations(ctx context.Context,
 	return nil
 }
 
+// CreateOAuth2mTLSDestinations is responsible to create an oauth2 mTLS destination resource in the remote destination service as well as in our DB
+func (s *Service) CreateOAuth2mTLSDestinations(ctx context.Context, destinationsDetails []operators.Destination, oauth2mTLSCredentials *operators.OAuth2mTLSAuthentication, formationAssignment *model.FormationAssignment, correlationIDs []string, skipSubaccountValidation bool) error {
+	for _, destinationDetails := range destinationsDetails {
+		if err := s.createOAuth2MTLSDestinations(ctx, destinationDetails, oauth2mTLSCredentials, formationAssignment, correlationIDs, skipSubaccountValidation); err != nil {
+			return errors.Wrapf(err, "while creating oauth2 mTLS destination with name: %q", destinationDetails.Name)
+		}
+	}
+	return nil
+}
+
+// createOAuth2MTLSDestinations is responsible to create an oauth2 mTLS destination resource in the remote destination service as well as in our DB
+func (s *Service) createOAuth2MTLSDestinations(ctx context.Context, destinationDetails operators.Destination, oauth2mTLSCredentials *operators.OAuth2mTLSAuthentication, formationAssignment *model.FormationAssignment, correlationIDs []string, skipSubaccountValidation bool) error {
+	subaccountID, err := s.destinationCreatorSvc.DetermineDestinationSubaccount(ctx, destinationDetails.SubaccountID, formationAssignment, skipSubaccountValidation)
```

Comment:
Instead of `s.destinationCreatorSvc.DetermineDestinationSubaccount` here we need to use the `EnsureDestinationSubaccountIDsCorrectness` method in the outside method(`CreateOAuth2mTLSDestinations`, the exported one) similar to the SAMLAssertion and ClientCertAuth destinations.
What is the reason for that: for these destinations, we have certificates and during certificate creation, we use this `EnsureDestinationSubaccountIDsCorrectness` method to ensure the subaccount is only one and we create the certificate/key store in that subaccount. After that during destination creation we need to ensure the subaccount is again only one. Because it will try to find the certificate/key store in that subaccount. If for some reason the subaccount is one but not the correct one, then it will fail, which is fine.